lib/scope.js

```diff
@@ -299,8 +301,10 @@ module.exports = {
   },
 
   load: async function ( scope, file_name, attempt_num ) {
-    /* Try to go to the interview url a few times with a pause in between each try. */
-    let max_attempts = 3;
+    /* Try to go to the interview url a few times */
+    let attempt_timeout = 10 * 1000;
+    // End failed attempts a little before full timeout so that we can get a useful error message
+    let max_attempts = scope.timeout/attempt_timeout - 100;
```

Comment:
What's the reason for the `- 100`? 

- `scope.timeout = 30 * 1000` (Is this right?)
- `attempt_timeout = 10 * 1000`
- `max_attempts = 3 - 100`

Reply:
Amazing catch!

> What's the reason for the `- 100`?

I was hoping the comment would make that clear, so once we figure out a good description, we should improve the comment. It's there because a (very) little time will have passed before the code got here. That means that if we use the exact `scope.timeout` in the async processes in the loops to follow and all loops get triggered, the Step itself will timeout before this whole process finishes. That'll cause a generic timeout error, not our custom error.

That comment doesn't help the code work, though, if the math in general is wrong.

> * `scope.timeout = 30 * 1000` (Is this right?)

That's the default timeout we've set, yes.

> * `attempt_timeout = 10 * 1000`

No note on this line other than to say I'm still kind of doubting my decision to only give the page 10 seconds to load. Would love an opinion on that, but maybe that's a question for the da crew.

> * `max_attempts = 3 - 100`

You are totally right. I forgot the parens around `attempt_timeout - 100`!

Also, proper math would result in 3.03 repeating, which does highlight the fact that `Math.floor()` is missing from here and so the whole thing would not have worked even if the parens would have been there. Which is why I wish I knew a way to test this properly gorrammit. It would require the server to be so slow that it doesn't load in time, though. I don't know how to do that, but if I did then the tests for this framework would slow the server way down whenever they were run.

Thanks for the catch!

Reply:
Aah, I think the math error was the issue for me. Now that I know it should have parens, everything makes sense. 

Re: being able to test sllllllllllllow servers, did you look into this already? https://stackoverflow.com/questions/27881485/network-throttling-with-chrome-and-selenium Selenium allows simulating slow internet speeds. It's not technically the same as a slow server, but it'll still make the page load more slowly.